rollup/sync_service/sync_service.go

```diff
@@ -139,13 +142,31 @@ func (s *SyncService) Stop() {
 	}
 }
 
+// ResetToHeight resets the SyncService to a specific L1 block height
+func (s *SyncService) ResetToHeight(height uint64) {
+	if s == nil {
+		return
+	}
+
+	s.stateMu.Lock()
+	defer s.stateMu.Unlock()
+
+	rawdb.WriteSyncedL1BlockNumber(s.db, height)
```

Comment:
I don't think you need to do this, no? `flush()` in `fetchMessages`will do that anyways.

Reply:
Probably same applies to `(s *RollupSyncService) ResetToHeight` as well

Reply:
changed in [75413e6](https://github.com/scroll-tech/go-ethereum/pull/1044/commits/75413e676b59f6f528436ab3282dbefbf2d47e76).